Remove Scalar Auto Refs -- Pass1

amirroth opened the pull request:
In parallel with removing bad uses of plain `auto`, I am also trying to eliminate some bad `auto &` idioms. One of these is the `auto &` to a scalar variable, e.g., `Real64` or int. `auto` aside (scalar variables should not be `auto`), these should be value shortcuts if they are not written to. If they are written to, then no shortcut should be used at all to avoid confusion with local variables. Here is a [longer explanation](https://github.com/NREL/EnergyPlus/wiki/Local-Shortcut-Variables:-When-to-Use-a-Value-Copy-and-When-to-Use-a-Reference). This initial pass tackles about 120 instances of the variables `state.dataHVACGlobal->TimeStepSys` and `state.dataHVACGlobal->SysElapsedTime`.

Question: CI comes back green, but there are a bunch of small diffs all over the place. I looked at three or four of them (there are about 150) and they are differences in the last decimal place of some meter or another. I am not sure if if it is worth bisecting these away.

amirroth committed:
Replaced about 120 instance of "auto &TimeStepSys" and "auto &SysTimeElapsed" with (non-ref) "Real64 TimeStepSys" and "Real64 SysTimeElapsed" or eliminated the shortcut and used "state.dataHVACGlobal->TimeStepSys" or "state.dataHVACGlobal->SysTimeElapsed" as appropriate.

amirroth committed:
Attempted fix for a failed regression test

amirroth commented on src/EnergyPlus/DemandManager.cc:
```diff
@@ -207,7 +207,7 @@ void SimulateDemandManagerList(EnergyPlusData &state,
 
     // Using/Aliasing
-    auto &TimeStepSys = state.dataHVACGlobal->TimeStepSys;
+    Real64 TimeStepSysSec = state.dataHVACGlobal->TimeStepSys * DataGlobalConstants::SecInHour;
```
The one other change I made was to create a shortcut for this common idiom, either instead of or in addition to shortcutting `TimeStepSys` itself.

JasonGlazer replied:
@amirroth I like the wider use of TimeStepSysSec but in a bunch of places it is surrounded by parenthesizes that are now unnecessary and reduce readability.

mjwitte replied:
How about adding `TimeStepSysSec` to `state` and compute it only once at the same time `TimeStepSys` is set (which could be more than one place due to downstepping).

amirroth replied:
> @amirroth I like the wider use of TimeStepSysSec but in a bunch of places it is surrounded by parenthesizes that are now unnecessary and reduce readability.

You've uncovered my secret which is that I did this largely using search-and-replace. I can search for this pattern and replace it also.

amirroth replied:
> How about adding `TimeStepSysSec` to `state` and compute it only once at the same time `TimeStepSys` is set (which could be more than one place due to downstepping).

I thought of doing this, but I was afraid I wouldn't find all the places in which `TimeStepSys` is modified (It had already bitten me once). I can give it a shot.

amirroth replied:
> > How about adding `TimeStepSysSec` to `state` and compute it only once at the same time `TimeStepSys` is set (which could be more than one place due to downstepping).
> 
> I thought of doing this, but I was afraid I wouldn't find all the places in which `TimeStepSys` is modified (It had already bitten me once). I can give it a shot.

Is `HVACManager.cc` the only place where this happens?

mjwitte replied:
> Is `HVACManager.cc` the only place where this happens?

Yes.

rraustad commented on src/EnergyPlus/DisplacementVentMgr.cc:
```diff
@@ -516,7 +516,8 @@ void CalcUCSDDV(EnergyPlusData &state, int const ZoneNum) // Which Zonenum
     using namespace DataHeatBalance;
-    auto &TimeStepSys = state.dataHVACGlobal->TimeStepSys;
+    Real64 TimeStepSys = state.dataHVACGlobal->TimeStepSys;
+    Real64 TimeStepSysSec = TimeStepSys * DataGlobalConstants::SecInHour;
```
TimeStepSys is only used here. Wouldn't it be better to use:

    Real64 TimeStepSysSec = state.dataHVACGlobal->TimeStepSys * DataGlobalConstants::SecInHour;

rraustad replied:
Nope, I see it at line 581.

amirroth replied:
A value shortcut is always fine even if it is only used once, because: i) this is essentially what the compiler does itself (it has to evaluate every expression piece by piece so each sub-expression essentially gets a register shortcut) and ii) you are not adding a pointer or reference for the compiler to keep track of and which may interfere with other optimizations.

rraustad commented on src/EnergyPlus/LowTempRadiantSystem.cc:
```diff
@@ -5346,17 +5346,14 @@ namespace LowTempRadiantSystem {
             int surfNum = this->SurfacePtr(radSurfNum);
 
-            if (state.dataLowTempRadSys->LastSysTimeElapsed(surfNum) == SysTimeElapsed) {
+            if (state.dataLowTempRadSys->LastSysTimeElapsed(surfNum) == state.dataHVACGlobal->SysTimeElapsed) {
```
Should use local here.

amirroth committed:
A little cleanup per review suggestions

amirroth committed:
Address comments.  Update/clean up unit tests.

amirroth commented:
Addressed comments. @mjwitte's suggestion to store `TimeStepSys * SecInHour` in `dataHVACGlobal` when `TimeStepSys` is updated necessitated updates to unit tests which do not call `HVACManager`. 

Still unable to run the correct version of `clang-format` locally.

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into RemoveScalarAutoRefs

Myoldmopar committed:
Clang formatting

Myoldmopar commented:
Applied clang format, I'm testing it locally here, and if everything seems to come back clean here we don't need to wait on another round of Decent.

Myoldmopar committed:
Disable maybe unneeded plugin test to cleanup CI

Myoldmopar commented:
While testing, I encountered a unit test failure (#9870).  It's unrelated to this change, as it is already in develop.  Since I'm about to merge this branch anyway, I'm going to just DISABLE the test here while we decide if the test is even necessary anymore.   The test suite is now passing fine with deda533 in place.  I won't wait on Decent CI, just merging this in.  Thanks @amirroth and reviewers.
